server/src/controllers/game.controller.js

```diff
@@ -375,7 +375,63 @@ class GameController {
   // Get Game By Id
   // async getById(req, res) {
   // }
+  
+  // send message in game chat
+  async message(req, res) {
+    const { message, game_id, user_id } = req.body; // user_id to be extracted from token
+    
+    // find user in db with decoded token (to be implemented later)
+    if (!user_id) {
+      throw new CustomError("Unauthorized access", 402);
+    }
+    
+    // find game in db
+    const gameExist = await GameRepo.fetchOne(game_id);
+    if (!gameExist.data) {
+      throw new CustomError("Game instance not found", 404);
+    }
 
+    // players should not be able to send message (to be implemented later)
+    
+    // incase user gets sloppy
+    const formattedMessage = message.trim();
+    if (!formattedMessage) {
+      throw new CustomError("message text cannot be empty", 400);
+    }
+
+    // test usernames and image_url
+    const testUsernames = ["mark", "jack", "jane", "crystal"];
+    const testImageUrl = "https://www.gravatar.com/avatar/00000000000000000000000000000000?d=robohash";
+    const random = Math.floor(Math.random() * testUsernames.length);
+
+    const messageProps = {
+         text: formattedMessage,
+         user_name: testUsernames[random],
+         image_url: testImageUrl // user_name & image_url from user info retrieved from db
+       };
+    
+    // push to message collection
+    // fetch message collection of game instance, if none create else update
+    if (!gameExist.data[0].messages) {
+      gameExist.data[0].messages = [];
+    }
+    gameExist.data[0].messages = 
+      gameExist.data[0].messages.concat(messageProps);
+
+    const response = await GameRepo.update(game_id, gameExist.data[0]);
+    
+    console.log(response);
+    // publish to centrifugo
+    await centrifugoController.publish(game_id, {
```

Comment:
This is not the publishing pattern. Look at other routes publishing to Centrifugo…

Reply:
what's the publishing pattern then? asides from not assigning it to a payload variable.